test/functional/core/tailable_cursor.test.js

```diff
@@ -1,42 +1,61 @@
 'use strict';
 const { expect } = require('chai');
-const { setupDatabase, withClientV2 } = require('../shared');
+const { runLater } = require('../../tools/utils');
 
 describe('Tailable cursor tests', function () {
-  before(function () {
-    return setupDatabase(this.configuration);
-  });
-
-  it('should correctly perform awaitData', {
-    metadata: { requires: { mongodb: '>=3.2' } },
-    test: withClientV2((client, done) => {
-      const db = client.db();
-      db.collection('cursor_tailable').drop(() => {
-        db.createCollection('cursor_tailable', { capped: true, size: 10000 }, (err, coll) => {
-          expect(err).to.not.exist;
+  describe('awaitData', () => {
+    let client;
 
-          coll.insertOne({ a: 1 }, (err, res) => {
-            expect(err).to.not.exist;
-            expect(res).property('insertedId').to.exist;
+    beforeEach(async function () {
+      client = this.configuration.newClient();
```

Comment:
do you think we'll continue using the `this.configuration.newClient` instead of our actual `MongoClient` in our tests? I don't think it does anything special in this case, and I think overall it does more harm by obscuring the actual setup than help, but I'm open to being convinced otherwise

Reply:
probably out of scope for this work; I think eventually we'd want this reworked into something like `new MongoClient( ...defaultOptions )` where the default options are more descriptively named (e.g., defaultOptionsWithVersionedAPI or something), so that we're actually passing in the stuff that matters, and it's clear what that stuff that matters is.